core-plugins/docs/XMLReader-batchsource.md

```diff
@@ -0,0 +1,107 @@
+# XMLReader Batch Source
+
+
+Description
+-----------
+The XML Reader plugin is a source plugin that allows users to read XML files stored on HDFS.
+
+
+Use Case
+--------
+Customer XML feeds that are dropped onto HDFS. These can be small to very large XML  files. The files have to read,
+parsed and when used in conjunction with XML Parser they are able to extract fields. This reader will emit one XML
+event that is specified by a Node Path to be extracted from the file.
+
+
+Properties
+----------
+**referenceName:** This will be used to uniquely identify this source for lineage, annotating metadata, etc.
+
+**path:** Path to file(s) to be read. If a directory is specified, terminate the path name with a '/'.
+
+**nodePath:** Node path to emit individual event from the schema.
+Example - '/book/price' to read only price under the book node
+
+**pattern:** Pattern to select specific file(s).
+Example -
+1. Use '^' to select file with name start with 'catalog', like '^catalog'.
+2. Use '$' to select file with name end with 'catalog.xml', like 'catalog.xml$'.
+3. Use '*' to select file with name contains 'catalogBook', like 'catalogBook*'.
+
+**actionAfterProcess:** Action to be taken after processing of the XML file.
+Possible actions are -
+1. Delete from the HDFS.
+2. Archived to the target location.
+3. Moved to the target location.
+
+**targetFolder:** Target folder path if user select action after process, either ARCHIVE or MOVE. Target folder must be
+an existing directory.
+
+**reprocessingRequired:** Flag to know if file(s) to be reprocessed or not.
+
+**tableName:** Table name to keep track of processed file(s).
+
+**temporaryFolder** Existing hdfs folder path having read and write access to the current User, required to store
+temporary file(s) containing path of the processed xml file(s). These temporary file(s) will be read at the end of job
+to update file track table.
+
+Example
+-------
+This example reads data from folder "/source/xmls/" and emits XML records on the basis of node path
+"/catalog/book/title". It will generate structured record with schema as 'offset', 'fileName', and 'record'.
```

Comment:
It will generate structured records with fields 'offset', 'fileName', and 'record'.